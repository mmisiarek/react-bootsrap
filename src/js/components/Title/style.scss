@import "../../dna";

.root {
  background: $bgcolor;
}
