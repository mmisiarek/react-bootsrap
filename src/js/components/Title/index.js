import React from 'react';
import PropTypes from 'prop-types';

import style from './style.scss';

const Title = props => <div className={style.root}>Hello {props.name}</div>;

Title.propTypes = {
  name: PropTypes.string,
};

Title.defaultProps = {
  name: 'stranger',
};

export default Title;
