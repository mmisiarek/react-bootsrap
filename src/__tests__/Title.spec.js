import React from 'react';

import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Title from '../js/components/Title/index';

describe('Title component', () => {
  it('should render without throwing an error', () => {
    const wrapper = shallow(<Title />);
    expect(wrapper.contains(<div>Hello stranger</div>)).toBe(true);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render name when provided', () => {
    const wrapper = shallow(<Title name="John" />);
    expect(wrapper.contains(<div>Hello John</div>)).toBe(true);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
