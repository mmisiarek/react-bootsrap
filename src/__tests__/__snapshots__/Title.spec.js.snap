// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component should render name when provided 1`] = `
<div>
  Hello 
  John
</div>
`;

exports[`Title component should render without throwing an error 1`] = `
<div>
  Hello 
  stranger
</div>
`;
